packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/Tabs/tabs.js

```diff
@@ -0,0 +1,297 @@
+import React, {
+    Children,
+    useRef,
+    useEffect,
+    useCallback,
+    useState
+} from 'react';
+import {
+    Tabs as TabWrapper,
+    TabList,
+    Tab as TabHeader,
+    TabPanel
+} from 'react-tabs';
+import defaultClasses from './tabs.css';
+import { mergeClasses } from '../../../../../classify';
+import { arrayOf, number, oneOf, shape, string } from 'prop-types';
+
+/**
+ * Page Builder Tabs component.
+ *
+ * This component is part of the Page Builder / PWA integration. It can be consumed without Page Builder.
+ *
+ * @typedef Tabs
+ * @kind functional component
+ *
+ * @param {props} props React component props
+ *
+ * @returns {React.Element} A React component that displays a set of Tabs.
+ */
+const Tabs = props => {
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const navigationRef = useRef(null);
+    const [isScrolling, setIsScrolling] = useState(false);
+    const [clientX, setClientX] = useState(0);
+    const [scrollX, setScrollX] = useState(0);
+    const [scrollElement, setScrollElement] = useState(null);
+    const [navWrapperClass, setNavWrapperClass] = useState();
```

Comment:
Rather than keeping classnames in state, you should keep local strings in state and retrieve the associated classname only at the very end as you're creating elements.

```jsx
const [gradient, setGradient] = useState("")

// note that this effect no longer depends on `classes`
useEffect(() => {
  // here, "BOTH" is the *actual state*
  // classname isn't state, it's a derivative of state
  setGradient("BOTH")
}, [setGradient])

// consider making a mapping outside of the component definition
// similar to how you look up flex properties
return (
  <div className={gradientClasses.get(gradient)}>
)
```

Reply:
This is a really neat solution, thanks for suggesting it :)

Reply:
Resolved: https://github.com/magento/pwa-studio/pull/1872/commits/dc8fb48bc7c2edba5bd3573da6e9fa28e48adfc1

Reply:
Thanks for implementing it. 😄